### PR TITLE
Feature/user model

### PR DESCRIPTION
**Pull request by Amira578**
[Added] Delete operation to userController
[Added] Request to handle validation 
[Added] Exception function to handle json response

### CODE REVIEW

File: app/Http/Controllers/API/V1/ProjectController.php [hunk 1]
```diff
@@ -8,17 +8,21 @@
...
+    {
         return Project::all();
     }
-    public function show( Project $project){
+    public function show( Project $project)
+    {
         return $project;
     }
-    public function store( Request $request){
+    public function store( Request $request)
```
Comment (aelnemr) [inline app/Http/Controllers/API/V1/ProjectController.php]: Request Form Validations?

File: app/Http/Controllers/API/V1/ProjectController.php [hunk 2]
```diff
@@ -8,17 +8,21 @@
...
+    {
         $project= Project::create($request->all());
         return $project;
     }
-    public function update( Request $request, Project $project){
+    public function update( Request $request, Project $project)
```
Comment (aelnemr) [inline app/Http/Controllers/API/V1/ProjectController.php]: Request Form Validations?

File: app/Http/Controllers/API/V1/TagController.php [hunk 3]
```diff
@@ -7,17 +7,21 @@
...
+{
     return Tag::all();
 }
-    public function show( Tag $tag){
+    public function show( Tag $tag)
+    {
         return $tag;
     }
-    public function store( Request $request){
+    public function store( Request $request)
```
Comment (aelnemr) [inline app/Http/Controllers/API/V1/TagController.php]: Request Form Validations?

File: app/Http/Controllers/API/V1/TagController.php [hunk 4]
```diff
@@ -7,17 +7,21 @@
...
+    {
         $tag= Tag::create($request->all());
         return $tag;
     }
-    public function update( Request $request, Tag $tag){
+    public function update( Request $request, Tag $tag)
```
Comment (aelnemr) [inline app/Http/Controllers/API/V1/TagController.php]: Request Form Validations?

File: app/Http/Controllers/API/V1/TaskController.php [hunk 5]
```diff
@@ -8,17 +8,21 @@
...
+    {
         return Task::all();
     }
-    public function show( Task $task){
+    public function show( Task $task)
+    {
         return $task;
     }
-    public function store( Request $request){
+    public function store( Request $request)
```
Comment (aelnemr) [inline app/Http/Controllers/API/V1/TaskController.php]: Request Form Validations?

File: app/Http/Controllers/API/V1/TaskController.php [hunk 6]
```diff
@@ -8,17 +8,21 @@
...
+    {
         $task= Task::create($request->all());
         return $task;
     }
-    public function update( Request $request, Task $task){
+    public function update( Request $request, Task $task)
```
Comment (aelnemr) [inline app/Http/Controllers/API/V1/TaskController.php]: Request Form Validations?

File: app/Http/Controllers/API/V1/TaskController.php [hunk 7]
```diff
@@ -8,17 +8,21 @@
 
 class TaskController extends Controller
 {
-    public function index(){
+    public function index()
+    {
         return Task::all();
```
Comment (aelnemr) [inline app/Http/Controllers/API/V1/TaskController.php]: Response resource?

File: app/Http/Controllers/API/V1/ProjectController.php [hunk 8]
```diff
@@ -8,17 +8,21 @@
 
 class ProjectController extends Controller
 {
-    public function index(){
+    public function index()
+    {
         return Project::all();
```
Comment (aelnemr) [inline app/Http/Controllers/API/V1/ProjectController.php]: Response resource?

**Review by aelnemr — CHANGES_REQUESTED** [review]
Please add Request Form Validations for all `store`, `update` functions and response resource for all returned objects and lists

File: app/Http/Controllers/API/V1/TagController.php (hunk 3 above)
Comment (Amira578) [reply app/Http/Controllers/API/V1/TagController.php]: still didn't do any resources or requests for task and tag models

**Review by aelnemr — APPROVED** [review]

**Pull request merged**